[CI] URLBlobContainerRetriesTests testReadBlobWithPrematureConnectionClose failing

**Build Scans:**
- [elasticsearch-periodic #4576 / openjdk23_checkpart1_java-matrix](https://gradle-enterprise.elastic.co/s/zv7tw2c72zjzw)
- [elasticsearch-periodic #4566 / openjdk23_checkpart1_java-matrix](https://gradle-enterprise.elastic.co/s/mpkcndvjm6quq)

**Reproduction Line:**
```
./gradlew ':modules:repository-url:test' --tests "org.elasticsearch.common.blobstore.url.URLBlobContainerRetriesTests.testReadBlobWithPrematureConnectionClose" -Dtests.seed=C16E7AC27C5A306B -Dtests.locale=wae -Dtests.timezone=America/Nassau -Druntime.java=23
```

**Applicable branches:**
8.15

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.common.blobstore.url.URLBlobContainerRetriesTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testReadBlobWithPrematureConnectionClose),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: ((a string containing "premature end of chunk coded message body: closing chunk expected" or a string containing "premature end of content-length delimited message body") or a string containing "connection closed prematurely")
     but: was "the target server failed to respond"
```

**Issue Reasons:**
- [8.15] 2 failures in test testReadBlobWithPrematureConnectionClose (1.9% fail rate in 104 executions)
- [8.15] 2 failures in step openjdk23_checkpart1_java-matrix (66.7% fail rate in 3 executions)
- [8.15] 2 failures in pipeline elasticsearch-periodic (50.0% fail rate in 4 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/test/framework/src/main/java/org/elasticsearch/repositories/blobstore/AbstractBlobContainerRetriesTestCase.java
+++ b/test/framework/src/main/java/org/elasticsearch/repositories/blobstore/AbstractBlobContainerRetriesTestCase.java
@@ -402,7 +402,9 @@ public abstract class AbstractBlobContainerRetriesTestCase extends ESTestCase {
         if (bytesToSend > 0) {
             exchange.getResponseBody().write(bytes, rangeStart, bytesToSend);
         }
-        if (randomBoolean()) {
+        if (randomBoolean() || Runtime.version().feature() >= 23) {
+            // For now in JDK23 we need to always flush. See https://bugs.openjdk.org/browse/JDK-8331847.
+            // TODO: remove the JDK version check once that issue is fixed
             exchange.getResponseBody().flush();
         }
         return bytesToSend;

